File: 배민/웹프론트엔드/01.js
function solution(arr) {

  var answer = -1;
  var numbers = arr
  while (true) {
    if (numbers.length === 1) {
      answer++;
      break
    }
    console.log(numbers)
    console.log(numbers.length)
    let temp = new Array()
    let num = 0;
    let seq = 0;
    for (let i = 0; i < numbers.length; i++) {
      let n = numbers[i];
      if (i == numbers.length-1) {
        if (num == n) {
          temp.push(++seq)
        } else {
          temp.push(seq)
          temp.push(1)

        }
        break;
      }
      else if (num == 0) {
        num = n;
        seq++;
      } else if (num == n) {
        seq++;
      } else {
        temp.push(seq)
        seq = 1;
        num = n;
      }
    }
    answer++;
    numbers = temp;
  }
  return ++answer;
}

// console.log(solution([1,1,3,3,2,2,4,5,1,1,1,3,3,3])) // 6
// console.log(solution([1,2,3])) // 3
// console.log(solution([2, 2, 1, 1, 2, 2, 1, 1, 2, 2, 2, 1, 2] )) // 5
console.log(solution([2])) // 1

File: 쿠팡/2020/01.py
# def convert(num, flag):
#     t = "0123456789ABCDEF"
#     q, r = divmod(num, flag)
#
#     if not q:
#         return t[r]
#     return convert(q, flag) + t[r]
#
#
# print(convert(2, 2))
#
# def solution(N):
#     answer = []
#     for i in range(2, 9):
#         result = convert(N, i)
#         tmp = 1
#         for n in str(result):
#             if n != '0':
#                 tmp *= int(n)
#         if not answer or answer[1] <= tmp:
#             answer = [i, tmp]
#         else:
#             if answer[1] <= tmp:
#                 answer = [i, tmp]
#     return answer
#
#
# print(solution(1000000))
#
# c

arr = [1, 2, 3]
print(sorted(arr, key=lambda x:(-x)))